tools/TestFx/Live/SaveLiveTestResult.ps1

```diff
@@ -17,60 +17,15 @@ param (
 
     [Parameter(Mandatory, Position = 4)]
     [ValidateNotNullOrEmpty()]
-    [string] $ClusterRegion,
-
-    [Parameter(Mandatory, Position = 5)]
-    [ValidateNotNullOrEmpty()]
-    [string] $DatabaseName,
-
-    [Parameter(Mandatory, Position = 6)]
-    [ValidateNotNullOrEmpty()]
-    [string] $TableName,
-
-    [Parameter(Mandatory, Position = 7)]
-    [ValidateNotNullOrEmpty()]
-    [string] $DataLocation,
-
-    [Parameter(Mandatory, Position = 8)]
-    [ValidateNotNullOrEmpty()]
-    [string] $BuildId,
-
-    [Parameter(Mandatory, Position = 9)]
-    [ValidateNotNullOrEmpty()]
-    [string] $OSVersion,
-
-    [Parameter(Position = 10)]
-    [string] $Tag = [string]::Empty
+    [string] $ClusterRegion
 )
 
-$liveTestDir = Join-Path -Path $DataLocation -ChildPath "LiveTestAnalysis" | Join-Path -ChildPath "Raw"
-$liveTestResults = Get-ChildItem -Path $liveTestDir -Filter "*.csv" -File -ErrorAction SilentlyContinue | Select-Object -ExpandProperty FullName
-
-if (![string]::IsNullOrWhiteSpace($liveTestResults)) {
-    if (![string]::IsNullOrEmpty($Tag)) {
-        $exProps = @{ Tag = $Tag } | ConvertTo-Json -Compress
-    }
-
-    $liveTestResults | ForEach-Object {
-        (Import-Csv -Path $_) |
-        Select-Object `
-        @{ Name = "Source"; Expression = { "LiveTest" } }, `
-        @{ Name = "BuildId"; Expression = { $BuildId } }, `
-        @{ Name = "OSVersion"; Expression = { $OSVersion } }, `
-        @{ Name = "PSVersion"; Expression = { $_.PSVersion } }, `
-        @{ Name = "Module"; Expression = { $_.Module } }, `
-        @{ Name = "Name"; Expression = { $_.Name } }, `
-        @{ Name = "Description"; Expression = { $_.Description } }, `
-        @{ Name = "StartDateTime"; Expression = { $_.StartDateTime } }, `
-        @{ Name = "EndDateTime"; Expression = { $_.EndDateTime } }, `
-        @{ Name = "IsSuccess"; Expression = { $_.IsSuccess } }, `
-        @{ Name = "Errors"; Expression = { $_.Errors } }, `
-        @{ Name = "ExtendedProperties"; Expression = { $exProps } } |
-        Export-Csv -Path $_ -Encoding utf8 -NoTypeInformation -Force
-    }
+$ltDir = Join-Path -Path ${env:DATALOCATION} -ChildPath "LiveTestAnalysis" | Join-Path -ChildPath "Raw"
+$ltResults = Get-ChildItem -Path $ltDir -Filter "*.csv" -File -ErrorAction SilentlyContinue | Select-Object -ExpandProperty FullName
 
+if (![string]::IsNullOrEmpty($ltResults)) {
     Import-Module "./tools/TestFx/Utilities/KustoUtility.psd1" -ArgumentList $ServicePrincipalTenantId, $ServicePrincipalId, $ServicePrincipalSecret, $ClusterName, $ClusterRegion -Force
-    Import-KustoDataFromCsv -DatabaseName $DatabaseName -TableName $TableName -CsvFile $liveTestResults
+    Import-KustoDataFromCsv -DatabaseName ${env:LIVETESTDATABASENAME} -TableName ${env:LIVETESTTABLENAME} -CsvFile $ltResults
```

Comment:
Import is a little confusing. consider publish

Reply:
@isra-fel Publish is still not that accurate. How about **Save-KustoDataFromCsv**? Or just **Save-KustoData**. Later on, this cmdlet can expand to save the data from multiple sources like json, storage via different parametersets. Your thoughts?

Reply:
Let's use `Add-KustoData`